snapcraft/internal/lifecycle.py

```diff
@@ -312,30 +311,11 @@ def _handle_dirty(self, part, step, dirty_report):
         part.clean(staged_state, primed_state, step, '(out of date)')
 
 
-def _create_tar_filter(tar_filename):
-    def _tar_filter(tarinfo):
-        fn = tarinfo.name
-        if fn.startswith('./parts/') and not fn.startswith('./parts/plugins'):
-            return None
-        elif fn in ('./stage', './prime', tar_filename):
-            return None
-        elif fn.endswith('.snap'):
-            return None
-        return tarinfo
-    return _tar_filter
-
-
 def cleanbuild(project_options, remote=''):
     config = snapcraft.internal.load_config(project_options)
-    tar_filename = '{}_{}_source.tar.bz2'.format(
```

Comment:
I just don't want the tar logic to move into `lxd`. Either duplicate code (which should be fine as it is temprorary) or implement the mount stuff here (it is a few lines after all.